index.js

```diff
@@ -115,7 +115,7 @@ discordClient.on('ready', () =>{
                     message.channel.send(botMessage);
                 }
                 else {
-                    message.channel.send("Weather request format is incorrect.")
+                    message.channel.send("U.S. ZIP code not detected (5 digits)")
                     return
                 }
                 
```

Comment:
Rather than outputting a specific message, I'd allow for a variety some of which are generated in the methods called by `getWeather`:

```javascript
        if(message.content.startsWith('!')) {
          try {
            const zip = message.content.slice(1).trim()
                
            if(!/^\d{5}$/.test(msg) ) {
             throw new Error('This program only accepts U.S. five digit zip codes.')
          }

          const { city, temperature } = await getWeather(zip)
          const out = `The current temperature in ${city} is ${temperature}°F.`
          message.channel.send(out)
        } catch(error) {
          message.channel.send(`Error: ${error.message}`)
        }
      }
```

You could check the return values from your API endpoints and throw errors within those methods and they will bubble up to here and get outputted.